package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -171,38 +204,38 @@ def _single_frame(self):
                                    bins=self.nbins,
                                    range=(0.0, max(self.dimensions)))
 
-            self.results[dim][key] += hist
-            self.results[dim][key_std] += np.square(hist)
+            self._results[f"{dim}_{key}"] += hist
+            self._results[f"{dim}_{key_std}"] += np.square(hist)
 
     def _conclude(self):
         k = 6.022e-1  # divide by avodagro and convert from A3 to cm3
 
         # Average results over the  number of configurations
         for dim in ['x', 'y', 'z']:
-            for key in ['pos', 'pos_std', 'char', 'char_std']:
-                self.results[dim][key] /= self.n_frames
+            for key in self._keys:
+                self._results[f"{dim}_{key}"] /= self.n_frames
```

Comment:
I see a bit of somewhat-complex discussion about backward compatibility in the matching issue. Maybe a summary of what is going on here would be useful?

You want to use `self.results` to hold an instance of the new common `_Results` class, so you are displacing the old use of `self.results`, but also changing its data structure?

I guess the usual questions about needed deprecations come up (depends on the policy adopted for MDA 2.0 I suppose).

Reply:
> You want to use self.results to hold an instance of the new common _Results class, so you are displacing the old use of self.results, but also changing its data structure?

I'm not sure if I got your question right but I'm always using the new data structure. Everything is saved to the attributes of the new _Results instance as defined in L145.

> I guess the usual questions about needed deprecations come up (depends on the policy adopted for MDA 2.0 I suppose).

 We already gave a warning that the structure of these modules will change in MDA 2.0 (PR https://github.com/MDAnalysis/mdanalysis/pull/3194)

Reply:
To be fair, that warning was only merged in 20 days ago.

Reply:
It looks like MDA core devs are in favor in any case--I'd probably lean a bit more conservatively, but fair enough.

Reply:
> I'm not sure if I got your question right but I'm always using the new data structure.

Ok, I think I found/still find the use of the alias with the underscore a bit confusing maybe.

Reply:
> To be fair, that warning was only merged in 20 days ago.

Sorry, for the delay but it took us a while to figure out how to structure the changes.

> It looks like MDA core devs are in favor in any case--I'd probably lean a bit more conservatively, but fair enough.

You would like to stick to dictionaries instead of named attributes? I love dictionaries if it is your own project and you are the only one accessing their items. If other people are involved these named attributes are convenient  allowing autocompletion is most IDEs.

> Ok, I think I found/still find the use of the alias with the underscore a bit confusing maybe.
 Write

As stated above, I'm totally fine with removing the alias.

Reply:
>  If other people are involved these named attributes are convenient allowing autocompletion is most IDEs.

I'm just saying that autocompletion on its own probably doesn't justify breaking user code without a pretty long/advertised deprecation period on the latest stable release. If you're writing a command line tool you could always write a shim layer to make it more convenient for your purposes.

I think the devs were "ok" with the breaking change though because of the bump to 2.0, so my nitpicking is probably irrelevant.

Reply:
I get your point and I'm usually also against breaking changes. Of course we could write a parser checking different  types but a universal attribute for all classes sounds reasonable for breaking some code - at least for me.

I don't think this is nitpicking. A RFC PR is made for being discussed :).